batch/batch/front_end/front_end.py

```diff
@@ -586,12 +600,39 @@ async def _get_full_job_status(app, record):
         raise
 
 
+# deprecated
 @routes.get('/api/v1alpha/batches/{batch_id}/jobs/{job_id}/log')
 @rest_billing_project_users_only
 async def get_job_log(request, userdata, batch_id):  # pylint: disable=unused-argument
     job_id = int(request.match_info['job_id'])
-    job_log = await _get_job_log(request.app, batch_id, job_id)
-    return web.json_response(job_log)
+    job_log_bytes = await _get_job_log(request.app, batch_id, job_id)
+    job_log_strings: Dict[str, Optional[str]] = {}
+    for container, log in job_log_bytes.items():
+        try:
+            job_log_strings[container] = log.decode('utf-8') if log is not None else None
+        except UnicodeDecodeError as e:
+            raise web.HTTPBadRequest(
+                reason=f'log for container {container} is not valid UTF-8, upgrade your hail version to download the log'
+            ) from e
+    return web.json_response(job_log_strings)
+
+
+async def get_job_container_log(request, userdata, batch_id):
```

Comment:
The checks are failing because `userdata` is not used. I think we don't need to use it because the decorators on the actual routes does the user validation check before it gets to this point. Would be great if you could double check that.

Reply:
Yup you're right it's used by the decorator and unused in the function I added.